core/src/main/java/org/elasticsearch/indices/analysis/PreBuiltTokenFilters.java

```diff
@@ -469,10 +314,16 @@ protected boolean isMultiTermAware() {
     protected final PreBuiltCacheFactory.PreBuiltCache<TokenFilterFactory> cache;
 
 
+    private final CachingStrategy cachingStrategy;
     PreBuiltTokenFilters(CachingStrategy cachingStrategy) {
+        this.cachingStrategy = cachingStrategy;
         cache = PreBuiltCacheFactory.getCache(cachingStrategy);
     }
 
+    public CachingStrategy getCachingStrategy() {
+        return cachingStrategy;
+    }
+
     private interface MultiTermAwareTokenFilterFactory extends TokenFilterFactory, MultiTermAwareComponent {}
 
     public synchronized TokenFilterFactory getTokenFilterFactory(final Version version) {
```

Comment:
I've preserved this method entirely for `PreBuiltAnalyer.LOWERCASE#getMultiTermComponent`. I'll figure out how to move away from that at some point but I thought that too should wait for another PR.